bin/main.dart

```diff
@@ -22,13 +22,22 @@ void main(List<String> args) async {
       exit(0);
     }
 
+    final env = Platform.environment;
+
+    final port = int.parse(env['TUDO_SERVER_PORT'] ?? result['port']);
+    final database = env['TUDO_SERVER_DB'] ?? result['db'];
+    final dbHost = env['TUDO_SERVER_DB_HOST'] ?? result['db-host'];
+    final dbPort = int.parse(env['TUDO_SERVER_DB_PORT'] ?? result['db-port']);
+    final dbUsername = env['TUDO_SERVER_DB_USERNAME'] ?? result['db-username'];
+    final dbPassword = env['TUDO_SERVER_DB_PASSWORD'] ?? result['db-password'];
+
     await TudoServer().serve(
-      port: int.parse(result['port']),
-      database: result['db'],
-      dbHost: result['db-host'],
-      dbPort: int.parse(result['db-port']),
-      dbUsername: result['db-username'],
-      dbPassword: result['db-password'],
+      port: port,
+      database: database,
+      dbHost: dbHost,
+      dbPort: dbPort,
+      dbUsername: dbUsername,
+      dbPassword: dbPassword,
```

Comment:
I'm not particularly familiar with Dart, so I would love to hear if there is a better way to handle the env vars!

Reply:
This looks fine, I'm not too familiar with using env vars in Dart either TBH.

Any particular reason to prioritize the env vars over command params?
I'd have done it the other way around since passing command parameters feels more intentional.

Reply:
I agree that would make more sense. I did it this way because it was easier as the args have defaults set, so just swapping the approach here wouldn't work, and I was a little bit lazy 😅 

In other languages I've used arg parsers that also support env variables directly. That would be the easiest approach, but would of course require swapping out the parser that is used. I can find one package that seems to support this, but I have no idea whether the approach that uses is appropriate here (maybe it's overkill)? https://pub.dev/packages/smart_arg

Reply:
I see what you mean about the default arguments.
I'd say let's leave it as it is. I generally try to keep external dependencies to a minimum, so let's just go with what we have.